packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -194,6 +194,17 @@ ruleTester.run('prefer-optional-chain', rule, {
     'match && match$1 !== undefined;',
     'foo !== null && foo !== undefined;',
     "x['y'] !== undefined && x['y'] !== null;",
+    // private properties
+    'this.#a && this.#b;',
+    '!this.#a || !this.#b;',
+    'a.#foo && a.#foo.bar;',
+    '!a.#foo || !a.#foo.bar;',
+    'a.#foo?.bar;',
+    '!a.#foo?.bar;',
+    'foo().#a;',
+    'a.b.#a;',
+    'new A().#b;',
+    '(await a).#b;',
```

Comment:
You need to make them actually trigger the visitor, i.e.

```suggestion
    '!foo().#a || a;',
    '!a.b.#a || a;',
    '!new A().#b || a;',
    '!(await a).#b || a;',
```